src/presolve/remove_ifix.jl

```diff
@@ -0,0 +1,154 @@
+# ̃xᵀ̃Hx̃ + ̃ĉᵀx̃ + lⱼ²Hⱼⱼ + cⱼxⱼ + c₀
+# ̂c = ̃c + 2lⱼΣₖHⱼₖxₖ , k ≂̸ j
+
+"""
+    xrm, c0ps, nvarrm = remove_ifix!(ifix, Hrows, Hcols, Hvals, nvar, 
+                                     Arows, Acols, Avals, c, c0, 
+                                     lvar, uvar, lcon, ucon)
+
+Remove rows and columns in H, columns in A, and elements in lcon and ucon
+corresponding to fixed variables, that are in `ifix`
+(They should be the indices `i` where `lvar[i] == uvar[i]`).
+
+Returns the removed elements of `lvar` (or `uvar`), the offset in the QP
+objective `c0ps`, and the new number of variables `nvarrm`.
+`Hrows`, `Hcols`, `Hvals`, `Arows`, `Acols`, `Avals`, `c`, `lvar`, `uvar`,
+`lcon` and `ucon` are updated in-place.
+"""
+function remove_ifix!(
+  ifix,
+  Hrows,
+  Hcols,
+  Hvals,
+  nvar,
+  Arows,
+  Acols,
+  Avals,
+  c::AbstractVector{T},
+  c0,
+  lvar,
+  uvar,
+  lcon,
+  ucon,
+) where {T}
+  
+  # assume Hcols is sorted
```

Comment:
No you're right to point that out. It is by default with this constructor though I think: https://github.com/JuliaSmoothOptimizers/QuadraticModels.jl/blob/2976db43c315dbdb062a41c6dc7debb139af8af7/src/qpmodel.jl#L89

Should I make a sort check on the other constructor that uses the COO coordinates as inputs, or is this only specific to the presolve (so make a check in the presolve even though if the QP is constructed with https://github.com/JuliaSmoothOptimizers/QuadraticModels.jl/blob/2976db43c315dbdb062a41c6dc7debb139af8af7/src/qpmodel.jl#L89 Hcols is already sorted) ?

Reply:
We could add a keyword arg to the constructor `sorted = false` so that we perform a solve by default. If the user knows that Hcols will be sorted (as is the case when Hcols results from a call to `findnz()`), they can set `sorted = true`.

Reply:
OK that's a good idea! I suggest we do this in another PR

Reply:
Ok. We should probably merge that other PR first and rebase this one.